_source/_posts/2022-04-07-where-okta-meets-devops.md

```diff
@@ -0,0 +1,66 @@
+
+## Accessing SaaS services and tooling
+
+In 2022, it's vanishingly rare for a tech company to host every software product that they use on- premises. Although there are open or [self-hosted versions](https://github.com/awesome-selfhosted/awesome-selfhosted) of every major tool in the modern workflow, maintaining separate services gets prohibitively expensive. Instead of hosting physical servers, most organizations rent cloud resources. Instead of running email ([Mailman](https://list.org/)), chat ([IRC](https://datatracker.ietf.org/doc/html/rfc1459), [XMPP](https://xmpp.org/about/technology-overview/)), user account management ([LDAP](https://ldap.com/)), and more in-house, companies instead rent services providing these features from specialist providers. 
+
+Before the pattern of using a central identity provider and protocols like OAuth became commonplace, users would have to manage a separate login and password to every software-as-a-service vendor that their company used. This created security and administrative headaches for everyone involved. But thanks to widespread adoption of the underlying protocols, a company can now use a single identity provider like Okta to allow everyone in the organization to access all of the software that it buys for them. 
+
+As well as the obvious workplace services like Slack and email, DevOps engineers might use their Okta accounts to sign in to cloud service providers like [AWS](https://www.okta.com/aws/), [GCP](https://www.okta.com/%20partners/google/), or [Azure](https://www.okta.com/integrations/azure-portal-login/#overview).
+
+In a CI/CD pipeline, one might use Okta to log in to [Jenkins](https://www.okta.com/integrations/jenkins/#overview), [Spinnaker](https://docs.armory.io/armory-enterprise/armory-admin/auth-okta-configure/), or even [CircleCI](https://www.okta.com/integrations/circleci/#overview). If an organization uses [GitHub Enterprise](https://saml-doc.okta.com/Provisioning_Docs/GitHub_Enterprise_Provisioning.html), DevOps personnel can use Okta to sign in there too. 
+
+## Automating Okta administration tasks
+Just as sysadmins and IT managed identity for small to medium organizations in the olden days, DevOps engineers may be responsible for some administrative tasks in their company's Okta instance. 
+
+Software and infrastructure administration and management tasks can be automated using a plethora of tools, including Hashicorp's Terraform. Terraform allows users to describe the desired state of their resources, and then changes the resources to match that description.  Using [Okta's Terraform provider](https://github.com/okta/terraform-provider-okta), users can manage their Okta resources and track their changes with version control. This is particularly useful for those whose business needs have them managing many different Okta accounts and trying to keep the settings consistent across all of them. 
```

Comment:
can okta be used to manage access to TF cloud?

Reply:
good catch, it looks like we can!

Reply:
I've added that to the discussion of using okta to login to SaaS, though, as answering "how does a customer manage access to modules which it publishes" right next to the question "how does a customer benefit from using the tf that okta publishes" seems like it could get confusing